File: src/main/java/toast/client/utils/MovementUtil.java
package toast.client.utils;

import net.minecraft.client.MinecraftClient;
import net.minecraft.network.packet.c2s.play.PlayerMoveC2SPacket;
import net.minecraft.util.math.Vec3d;

public class MovementUtil {
    private static MinecraftClient mc = MinecraftClient.getInstance();

    public static void lookPacket(double yaw, double pitch) {
        if(mc.player == null) return;
        mc.player.networkHandler.sendPacket(new PlayerMoveC2SPacket.LookOnly((float) yaw, (float) pitch, mc.player.onGround));
    }

    public static void lookClient(double yaw, double pitch) {
        if(mc.player == null) return;
        mc.player.yaw = (float) yaw;
        mc.player.pitch = (float) pitch;
    }

    public static void lookAt(Vec3d point, boolean packet) {
        if(mc.player == null) return;
        double lookx = mc.player.getX() - point.getX();
        double looky = mc.player.getY() - point.getY();
        double lookz = mc.player.getZ() - point.getZ();

        double len = Math.sqrt(lookx * lookx + looky * looky + lookz * lookz);

        lookx /= len;
        looky /= len;
        lookz /= len;

        double pitch = Math.asin(looky);
        double yaw = Math.atan2(lookz, lookx);

        //to degree
        pitch = pitch * 180.0 / Math.PI;
        yaw = yaw * 180.0 / Math.PI;
        yaw += 90f;
        if(packet) {
            lookPacket(yaw, pitch);
        } else {
            lookClient(yaw, pitch);
        }
    }

}

File: src/main/java/toast/client/utils/MovementUtil.kt
package toast.client.utils

import net.minecraft.client.MinecraftClient
import net.minecraft.network.packet.c2s.play.PlayerMoveC2SPacket.LookOnly
import net.minecraft.util.math.Vec3d
import kotlin.math.asin
import kotlin.math.atan2
import kotlin.math.sqrt

object MovementUtil {
    private val mc = MinecraftClient.getInstance()
    private fun lookPacket(yaw: Double, pitch: Double) {
        if (mc.player == null) return
        mc.player!!.networkHandler.sendPacket(
            LookOnly(
                yaw.toFloat(),
                pitch.toFloat(),
                mc.player!!.onGround
            )
        )
    }

    private fun lookClient(yaw: Double, pitch: Double) {
        if (mc.player == null) return
        mc.player!!.yaw = yaw.toFloat()
        mc.player!!.pitch = pitch.toFloat()
    }

    fun lookAt(point: Vec3d, packet: Boolean) {
        if (mc.player == null) return
        var lookx = mc.player!!.x - point.getX()
        var looky = mc.player!!.y - point.getY()
        var lookz = mc.player!!.z - point.getZ()
        val len = sqrt(lookx * lookx + looky * looky + lookz * lookz)
        lookx /= len
        looky /= len
        lookz /= len
        var pitch = asin(looky)
        var yaw = atan2(lookz, lookx)

        //to degree
        pitch = pitch * 180.0 / Math.PI
        yaw = yaw * 180.0 / Math.PI
        yaw += 90.0
        if (packet) {
            lookPacket(yaw, pitch)
        } else {
            lookClient(yaw, pitch)
        }
    }
}